pkg/services/ngalert/state/manager_private_test.go

```diff
@@ -270,7 +270,15 @@ func TestProcessEvalResults_StateTransitions(t *testing.T) {
 		for _, ts := range tss {
 			results := resultsAtTime[ts]
 			clk.Set(ts)
-			actual := st.ProcessEvalResults(context.Background(), ts, alertRule, results, systemLabels)
+			var statesToSend StateTransitions
+			actual := st.ProcessEvalResults(context.Background(), ts, alertRule, results, systemLabels, func(_ context.Context, states StateTransitions) {
+				statesToSend = states
+			})
+
+			// Expect all statesToSend to have a LastSentAt set to the evaluation time.
+			for _, state := range statesToSend {
+				assert.Equalf(t, ts, *state.LastSentAt, "LastSentAt should be set to the evaluation time for all ready-to-send transitions.")
+			}
```

Comment:
Nit: this test suite is already pretty heavy and it focuses on testing the state transitions and permutations. Also, the tests assert LastSentAt. I would love to see a separate test that tests "send" functionality. WDYT? (If we agree on that, we can address it in a follow-up PR)